### PR TITLE
[a11y] 11.1.4.3 Contrast (minimum)

### PR DESCRIPTION
**Pull request by Aitorbp**
## Related Issues
App:

- [x] Add changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
- [x] Add feature to Release Notes in `ReleaseNotesViewModel.kt` creating a new `ReleaseNote()` with String resources (if required)

_____

## QA

https://github.com/owncloud/android/pull/4429#issuecomment-2205785319

### CODE REVIEW

**Comment by Aitorbp** [comment]
Everything related to the toolbar will be done in this issue: https://github.com/owncloud/android/issues/4365.
The avatar change does not apply to Android.

**Comment by Aitorbp** [comment]
The `list_item_lastmod_and_filesize_text` color has been changed to meet minimum contrast requirements. This color is present in the main file list of the app and also appears in:

- Log list (subtitle): Log size
- Transfer list (subtitles and icons)
- Subtitle bottom sheet file

FYI @jesmrec

File: owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/FileListAdapter.kt [hunk 1]
```diff
@@ -337,7 +337,10 @@ class FileListAdapter(
                 (view.itemView.layoutParams as StaggeredGridLayoutManager.LayoutParams).apply {
                     isFullSpan = true
                 }
-                view.binding.footerText.text = file.text
+                view.binding.footerText.apply {
+                    text = file.text
+                    setTextColor(ContextCompat.getColor(context, R.color.accessibility_grey_color))
```
Comment (JuancaG05) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/FileListAdapter.kt:342]: Why doing it programatically?

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt [hunk 2]
```diff
@@ -300,6 +300,7 @@ abstract class DrawerActivity : ToolbarActivity() {
         val account = drawerViewModel.getCurrentAccount(this) ?: return
         drawerViewModel.getStoredQuota(account.name)
         drawerViewModel.userQuota.observe(this) { event ->
+            getAccountQuotaText()?.setTextColor(getColor(R.color.accessibility_grey_color))
```
Comment (JuancaG05) [inline owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt:303]: Same, why doing it programmatically?

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Some changes required here @Aitorbp

**Comment by Aitorbp** [comment]
The `list_item_lastmod_and_filesize_text` color have been changed for this  hex value  `#707575 `to meet with the requirements:

list_item_lastmod_and_filesize_text



Also we have created a new color `share_warning_text` (`#616465`) for public link warning description to meet the requirements:

share_warning_text


Contrast checked used: https://webaim.org/resources/contrastchecker/

File: owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/FileListAdapter.kt (hunk 1 above)
Comment (Aitorbp) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/FileListAdapter.kt:342]: Changed to the XML: `list_footer`

File: owncloudApp/src/main/res/values/colors.xml [hunk 3]
```diff
@@ -28,8 +28,8 @@
     <color name="owncloud_blue">#1b223d</color>
     <color name="owncloud_blue_accent">#4E85C8</color>
     <color name="owncloud_blue_bright">#00ddff</color>
-
-    <color name="list_item_lastmod_and_filesize_text">@color/secondaryTextColor</color>
+    <color name="accessibility_grey_color">#616465</color>
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudApp/src/main/res/values/colors.xml:31]: `list_item_lastmod_and_filesize_text` changed by lighter grey

File: owncloudApp/src/main/res/layout/share_file_layout.xml [hunk 4]
```diff
@@ -63,6 +63,7 @@
                 android:layout_marginStart="4dp"
                 android:layout_toEndOf="@+id/shareFileIcon"
                 android:text="@string/placeholder_filesize"
+                android:textColor="@color/accessibility_grey_color"
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudApp/src/main/res/layout/share_file_layout.xml:66]: `accessibility_grey_color` reference deleted

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt (hunk 2 above)
Comment (Aitorbp) [reply owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt:303]: Added in XML `nav_drawer_footer`

**Review by JuancaG05 — APPROVED** [review]
LGTM

**Comment by jesmrec** [comment]
QA checks:

### date and size in list of files

Contrast: `#707575` vs `#FFFFFF` ➡️  4.68 > 4.50 ✅ 

### drawer footer (quota)

Contrast: `#707575` vs `#FFFFFF` ➡️  4.68 > 4.50 ✅ 

### land-drawer footer (quota)

Contrast: `#707575` vs `#FFFFFF` ➡️  4.68 > 4.50 ✅ 

### list footer (#files & #folders)

Contrast: `#707575` vs `#FFFFFF` ➡️  4.68 > 4.50 ✅ 

### shared file size 

Contrast: `#707575` vs `#FFFFFF` ➡️  4.68 > 4.50 ✅ 

### warning public link

Contrast: `#525757` vs `#E0E0E0` ➡️  5.56 > 4.50 ✅ 

### expiration date in public link 

Contrast: `#707575` vs `#FFFFFF` ➡️  4.68 > 4.50 ✅ 

### folder picker - target folder with no writing permissions

Contrast: `#525757` vs `#E0E0E0` ➡️  5.56 > 4.50 ✅ 

### folder picker - share from 3rd party app to folder with no writing permissions

Contrast: `#525757` vs `#E0E0E0` ➡️  5.56 > 4.50 ✅ 

### log size

Contrast: `#707575` vs `#FFFFFF` ➡️  4.68 > 4.50 ✅ 

### upload list: status, size, date, account, space and target folder

Contrast: `#707575` vs `#FFFFFF` ➡️  4.68 > 4.50 ✅ 

### bottom sheet: date and size

Contrast: `#707575` vs `#FFFFFF` ➡️  4.68 > 4.50 ✅ 

tested all views in portrait/landscape orientation.

Also, i took a look to the high contrast mode, which look is much better than before. Moving forward an accessible app step by step 💯 

This is approved from my side.

**Pull request merged**